indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamIndexTaskIOConfig.java

```diff
@@ -51,7 +53,8 @@ public SeekableStreamIndexTaskIOConfig(
       final Boolean useTransaction,
       final DateTime minimumMessageTime,
       final DateTime maximumMessageTime,
-      @Nullable final InputFormat inputFormat
+      @Nullable final InputFormat inputFormat,
+      @Nullable final Duration taskDuration // can be null for backward compabitility
```

Comment:
I think its good to have Duration and no need to convert back and forth and also confusion of adding `mins` to the variable. You think there are any issues with the schema deserializing ? (I see at least `Period` used, maybe we can switch to it if so )

Also, regarding variable name change, Do you think there would be a usecase of user editing it to a different value ? I would prefer to use `taskDuration` so that no one touches and would also avoid the need for docs (Also, I dont find this documented anywhere, can you point me to the file ?)